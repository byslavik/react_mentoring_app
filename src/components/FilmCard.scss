.filmCard {
  flex-grow: 1;
  flex-shrink: 0;
  flex-basis: calc(23% - 40px);
  margin: 20px;
  max-width: 25%;
}
.filmCard img {
  height: 294px;
  position: relative;
  z-index: 2;
}
.filmCard h1 {
  margin: 10px 0;
  font-size: 15px;
}
.filmCard h1 span {
  float: right;
  background: #fff;
  border: 1px solid #666;
  border-radius: 3px;
  padding: 3px;
}
.imageWrap {
  height: 294px;
  width: 205px;
  z-index: 0;
  background: #ccc;
  position: relative;
  overflow: hidden;
  text-align: center;

  &:before {
    content: 'No Poster';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: #666;
    z-index: 1;
    font-weight: bold;
  }
}