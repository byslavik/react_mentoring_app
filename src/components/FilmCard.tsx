import * as React from 'react';


export default class FilmCard extends React.Component {

    render() {
        return (<h1>React Mentoring programm App</h1>)
    }
}
