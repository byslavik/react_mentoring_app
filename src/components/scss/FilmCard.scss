.filmCard {
  flex-basis: calc(23% - 40px);
  flex-grow: 1;
  flex-shrink: 0;
  margin: 20px;
  max-width: 25%;
}
.filmCard img {
  height: 294px;
  position: relative;
  text-indent: -9999px;
  z-index: 2;
}
.filmCard h1 {
  font-size: 15px;
  margin: 10px 0;

  a {
      color: #000;
      text-decoration: none;

      &:hover {
          text-decoration: underline;
      }
  }
}
.filmCard h1 span {
  background: #fff;
  border: 1px solid #666;
  border-radius: 3px;
  float: right;
  padding: 3px;
}
.imageWrap {
  background: #ccc;
  height: 294px;
  overflow: hidden;
  position: relative;
  text-align: center;
  width: 205px;
  z-index: 0;

  &:before {
      color: #666;
      content: 'No Poster';
      font-weight: bold;
      left: 50%;
      position: absolute;
      top: 50%;
      transform: translate(-50%, -50%);
      z-index: 1;
  }
}
