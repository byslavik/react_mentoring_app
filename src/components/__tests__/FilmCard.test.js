import * as React from 'react';
import {FilmCard} from '../FilmCard.tsx';
import * as renderer from 'react-test-renderer';
import {filmData} from './testData'

jest.mock('react-router-dom',()=> ({Link:'Link'}));

describe('FilmCard component', ()=>{
  it('FilmCard component shows correctly', () => {
    const tree = renderer.create(
      <FilmCard item={filmData}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})