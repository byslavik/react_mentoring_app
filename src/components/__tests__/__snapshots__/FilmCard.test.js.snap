// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard component FilmCard component shows correctly 1`] = `
<article
  className={undefined}
>
  <div
    className={undefined}
  >
    <img
      alt="Title"
      src="https://image.tmdb.org/t/p/w500//somepath.jpg"
    />
  </div>
  <h1>
    <Link
      to="/film/Title/1"
    >
      Title
    </Link>
     
    <span>
      100500
    </span>
  </h1>
</article>
`;
